package com.itheima.service.impt;

import com.itheima.dao.DeptDao;
import com.itheima.pojo.Dept;
import com.itheima.service.DeptService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.List;

@Service
public class DeptServiceImpt implements DeptService {
    @Autowired
    private DeptDao deptDao;
    @Override
    public List<Dept> lines() {
//2. 对原始数据进行处理 , 组装部门数据
        List<String> lines = deptDao.lines();
        List<Dept> deptList = lines.stream().map(line -> {
            String[] parts = line.split(",");
            Integer id = Integer.parseInt(parts[0]);
            String name = parts[1];
            LocalDateTime updateTime = LocalDateTime.parse(parts[2], DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
            return new Dept(id, name, updateTime);
        }).toList();

        //2. 响应数据
        return deptList;
    }
}
